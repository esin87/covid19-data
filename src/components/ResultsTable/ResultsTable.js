import React from 'react';
import Table from 'react-bootstrap/Table';

const ResultsTable = ({ data, formatNumber }) => {
	return (
		<Table>
			<thead>
				<tr>
					<th></th>
					<th>Confirmed cases</th>
					<th>Deaths</th>
					<th>Recovered</th>
				</tr>
			</thead>
			<tbody>
				<tr>
					<td>Total</td>
					<td>{formatNumber(data.Confirmed)}</td>
					<td>{formatNumber(data.Deaths)}</td>
					<td>{formatNumber(data.Recovered)}</td>
				</tr>
			</tbody>
		</Table>
	);
};

export default ResultsTable;
