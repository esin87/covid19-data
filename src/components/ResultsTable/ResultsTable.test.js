import React from 'react';
import ResultsTable from './ResultsTable';
import { mount, render } from 'enzyme';

describe('ResultsTable component', () => {
	let component;
	const data = {
		Country: 'Switzerland',
		CountryCode: 'CH',
		Lat: '46.82',
		Lon: '8.23',
		Confirmed: 20505,
		Deaths: 666,
		Recovered: 6415,
		Active: 13424,
		Date: '2020-04-05T00:00:00Z',
		LocationID: '628d4f12-6ebe-4fa9-b046-77ff0b894a4e',
	};
	function formatNumber(num) {
		return num.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1,');
	}

	beforeEach(() => {
		component = mount(<ResultsTable data={data} formatNumber={formatNumber} />);
	});

	it('renders', () => {
		render(component);
	});

	it('should match the snapshot', () => {
		expect(component.html()).toMatchSnapshot();
	});

	it('should have the first table head row be empty', () => {
		expect(component.find('th').at(0).text()).toBe('');
	});

	it('should have a second table head row with the appropriate text', () => {
		expect(component.find('th').at(1).text()).toBe('Confirmed cases');
	});

	it('receives the appropriate props', () => {
		expect(component.props()).toEqual({
			data: {
				Active: 13424,
				Confirmed: 20505,
				Country: 'Switzerland',
				CountryCode: 'CH',
				Date: '2020-04-05T00:00:00Z',
				Deaths: 666,
				Lat: '46.82',
				LocationID: '628d4f12-6ebe-4fa9-b046-77ff0b894a4e',
				Lon: '8.23',
				Recovered: 6415,
			},
			formatNumber: expect.any(Function),
		});
	});
});
