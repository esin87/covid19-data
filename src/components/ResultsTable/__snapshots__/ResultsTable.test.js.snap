// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsTable component should match the snapshot 1`] = `"<table class=\\"table\\"><thead><tr><th></th><th>Confirmed cases</th><th>Deaths</th><th>Recovered</th></tr></thead><tbody><tr><td>Total</td><td>20,505</td><td>666</td><td>6,415</td></tr></tbody></table>"`;
